test/save_assemblies_from_fastas_integration_test.py

```diff
@@ -576,4 +600,138 @@ def test_fail_mismatched_inputs(config, impl, context, scratch):
+            {'file': tmp_dir / 'GCF_000970245.1_ASM97024v1_genomic.fna.gz', 'assembly_name': 'GCF_000970245.1_ASM97024v1_genomic.fna.gz'}
+        ]
+    }
+
+    results = impl.save_assemblies_from_fastas(context, params)[0]['results']
+    _check_result_object_info_fields(config, results, file_names, object_metas)
+
+
+def test_invalid_threads_param(config, context, scratch):
+    tmp_dir = scratch / ("test_invalid_threads_param" + str(uuid.uuid4()))
+    os.makedirs(tmp_dir)
+    data = Path('data')
+    file_names = [
+        'GCA_002506415.1_ASM250641v1_genomic.fna.gz',
+        'GCF_000007065.1_ASM706v1_genomic.fna.gz'
+    ]
+
+    # copy 2 assembly files into the data dir
+    for file_name in file_names:
+        shutil.copy(data / file_name, tmp_dir)
+
+    params = {
+        'workspace_id': config['ws_id'],
+        'inputs': [
+            {'file': tmp_dir / 'GCA_002506415.1_ASM250641v1_genomic.fna.gz', 'assembly_name': 'GCA_002506415.1_ASM250641v1_genomic.fna.gz'},
+            {'file': tmp_dir / 'GCF_000007065.1_ASM706v1_genomic.fna.gz', 'assembly_name': 'GCF_000007065.1_ASM706v1_genomic.fna.gz'}
+        ]
+    }
+
+    MAX_THREADS = "MAX_THREADS"
+    THREADS_PER_CPU = "THREADS_PER_CPU"
+    config_dict = config['file_config'].copy()
+
+    # max_threads type check fails
+    config_dict["KBASE_SECURE_CONFIG_PARAM_MAX_THREADS"] = "10.5"
+    config_dict["KBASE_SECURE_CONFIG_PARAM_THREADS_PER_CPU"] = "2.5"
+    with raises(Exception) as got:
+        AssemblyUtil(config_dict)
+    assert_exception_correct(got.value, ValueError(f"{MAX_THREADS} must be an integer"))
+
+    # threads_per_cpu type check fails
+    config_dict["KBASE_SECURE_CONFIG_PARAM_MAX_THREADS"] = "10"
+    config_dict["KBASE_SECURE_CONFIG_PARAM_THREADS_PER_CPU"] = "2.8e"
+    with raises(Exception) as got:
+        AssemblyUtil(config_dict)
+    assert_exception_correct(got.value, ValueError(f"{THREADS_PER_CPU} must be an integer or decimal"))
+
+    # max_threads input check fails
+    config_dict["KBASE_SECURE_CONFIG_PARAM_MAX_THREADS"] = "0"
+    config_dict["KBASE_SECURE_CONFIG_PARAM_THREADS_PER_CPU"] = "2"
+    impl = AssemblyUtil(config_dict)
+    with raises(Exception) as got:
+        impl.save_assemblies_from_fastas(context, params)
+    assert_exception_correct(got.value, ValueError(f"{MAX_THREADS} must be > 0"))
+
+    # threads_per_cpu input check fails
+    config_dict["KBASE_SECURE_CONFIG_PARAM_MAX_THREADS"] = "10"
+    config_dict["KBASE_SECURE_CONFIG_PARAM_THREADS_PER_CPU"] = "0"
+    impl = AssemblyUtil(config_dict)
+    with raises(Exception) as got:
+        impl.save_assemblies_from_fastas(context, params)
+    assert_exception_correct(got.value, ValueError(f"{THREADS_PER_CPU} must be > 0"))
+
+
+def test_generator_overflow(config, scratch):
+    tmp_dir = scratch / ("test_generator_overflow" + str(uuid.uuid4()))
```

Comment:
Could you add a quick description as to the purpose of this test? it may not be obvious to the next person to come along and work on this code.

Reply:
Updated with some desc and feel free to supplement.

Reply:
Updated with some desc and feel free to supplement.

Reply:
:+1: